### PR TITLE
[TRANSFORMATIONS] Exclude Gather from Mixed prc propagation

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Exclude Gather from sensitive prc types *
 - *Add handling of `Const fp16 ->Gather -> Pow` pattern*

### Tickets:
 - *https://jira.devtools.intel.com/browse/CVS-127678*

### CODE REVIEW

File: src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp [hunk 1]
```diff
@@ -361,6 +361,44 @@ class MarkDivWithEps : public MatcherPass {
     }
 };
 
+// todo: description
+class GatherHandling : public MatcherPass {
+public:
+    OPENVINO_RTTI("GatherHandling", "0");
```
Comment (pavel-esir) [inline src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp:367]: Do we need this transformation anymore? Looks like to solve the problem with extra RAM CVS-127678 it should be enough just to remove Gather/GatherBase from the list above.

Comment (iefode) [reply src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp:367]: Pow is type sensitive op. It means that we propagate fp32 to pow inputs, also gather became fp32. Transformation mark gather and it's inputs as eligible for fp16 compression. Check the test

Comment (itikhono) [reply src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp:367]: I've commented GatherHandling transformation 
image

and got the expected subgraph with fp16 Gather operation
image

the test passes if I comment these 2 lines 
image

Comment (iefode) [reply src/common/transformations/src/transformations/fp16_compression/mark_subgraphs_to_keep_in_mixed_precision.cpp]: Have removed

**Review by itikhono — APPROVED** [review]

**Pull request merged**